app/uk/gov/hmrc/exports/services/notifications/NotificationHelper.scala

```diff
@@ -0,0 +1,120 @@
+import uk.gov.hmrc.http.HeaderCarrier
+
+@Singleton
+class NotificationHelper @Inject()(
+  customsDataStoreConnector: CustomsDataStoreConnector,
+  emailConnector: EmailConnector,
+  notificationRepository: NotificationRepository,
+  submissionRepository: SubmissionRepository
+)(implicit executionContext: ExecutionContext)
+    extends Logging {
+
+  def parseAndSave(unparsed: Notification)(implicit hc: HeaderCarrier): Future[Unit] =
+    Future {
+      Try(XML.loadString(unparsed.payload)).map(NotificationParser.parse) match {
+        case Success(notificationDetails) if notificationDetails.nonEmpty =>
+          saveParsedNotifications(unparsed, notificationDetails)
+
+        case Success(_) => ()
+
+        case Failure(exc) =>
+          logger.warn(s"There was a problem while parsing notification with actionId=${unparsed.actionId}. Error was: ${exc.getMessage}")
+      }
+    }
+
+  private def sendEmailForDmsdocNotification(emailAddress: VerifiedEmailAddress, mrn: String, eori: String)(implicit hc: HeaderCarrier): Unit = {
+    val emailParameters = EmailParameters(Map(EmailParameter.MRN -> mrn))
+    val sendEmailRequest = SendEmailRequest(List(emailAddress.address), TemplateId.DMSDOC_NOTIFICATION, emailParameters)
+    emailConnector.sendEmail(sendEmailRequest).map {
+      case EmailAccepted                      =>
+      case BadEmailRequest(message)           => logSendEmailError("Bad request", message, eori, mrn)
+      case InternalEmailServiceError(message) => logSendEmailError("Email service error", message, eori, mrn)
+    }
+  }
+
+  private def sendEmailIfDmsdocNotification(details: NotificationDetails, submission: Submission)(implicit hc: HeaderCarrier): Boolean = {
+    if (details.status == SubmissionStatus.ADDITIONAL_DOCUMENTS_REQUIRED)
+      customsDataStoreConnector.getEmailAddress(submission.eori).map {
+        _.fold(logEmailAddressNotFound(submission.eori))(sendEmailForDmsdocNotification(_, details.mrn, submission.eori))
+      }
+
+    true
+  }
+
+  private def saveParsedNotification(notification: Notification)(implicit hc: HeaderCarrier): Future[Boolean] =
+    notificationRepository.add(notification).flatMap {
+      _.fold(logNotificationError(_, notification), _ => updateRelatedSubmission(notification))
+    }
+
+  private def saveParsedNotifications(notification: Notification, parsedDetails: Seq[NotificationDetails])(implicit hc: HeaderCarrier): Unit =
+    Future.sequence {
+      parsedDetails.map { details =>
+        saveParsedNotification(Notification(notification.actionId, notification.payload, Some(details)))
+      }
+    } map { outcomes =>
+      // IS THIS OK?
+      // Removing from the repo the unparsed notification only if all notifications
+      // derived from the former after the parsing were successfully saved.
+      if (outcomes.forall(identity)) notificationRepository.removeUnparsedNotificationsForActionId(notification.actionId)
```

Comment:
:thinking: Given the scenario where a Notification contained multiple details, but one of the details fails to be parsed, would this not result in the Notification being re-parsed on each app startup, and so the successful details being inserted into Mongo again and again (duplicated)?

Reply:
Indeed. On the other side, removing no matter what the "un-parsed" notification will eliminate the chance to re-parse it if one of the detail response is not successfully saved.
It's not that I'm in favor of one or the other solution. The current implementation is just to bring up the issue and discuss it.

Reply:
Current parser works in a way that if there is parsing error, no `NotificationDetails` are being returned. Instead, it throws exception so it's all-or-nothing approach. 
Therefore, there is no risk of having duplicated Notifications because of that.

Reply:
:thinking: The problem of the removal of the un-parsed notification is not related to the parsing outcome, isn't it? But to the outcome of storing the single/multiple parsed notifications as well as if the related Submission can be updated or not.